src/controllers/csi/provisioner/processmoduleconfig.go

```diff
@@ -74,14 +87,42 @@ func (provisioner *OneAgentProvisioner) readProcessModuleConfigCache(tenantUUID
 func (provisioner *OneAgentProvisioner) writeProcessModuleConfigCache(tenantUUID string, processModuleConfig *processModuleConfigCache) error {
 	processModuleConfigCache, err := provisioner.fs.OpenFile(provisioner.path.AgentRuxitProcResponseCache(tenantUUID), os.O_WRONLY|os.O_CREATE|os.O_TRUNC, 0644)
 	if err != nil {
+		log.Info("Error opening processModuleConfigCache, when writing")
+		provisioner.removeProcessModuleConfigCache(tenantUUID)
 		return err
 	}
+
 	jsonBytes, err := json.Marshal(processModuleConfig)
 	if err != nil {
 		processModuleConfigCache.Close()
+		log.Info("Error when marshaling processModuleConfigCache")
+		provisioner.removeProcessModuleConfigCache(tenantUUID)
+		return err
+	}
+
+	if _, err := processModuleConfigCache.Write(jsonBytes); err != nil {
+		processModuleConfigCache.Close()
+		log.Info("Error writing processModuleConfigCache")
+		provisioner.removeProcessModuleConfigCache(tenantUUID)
 		return err
 	}
-	_, err = processModuleConfigCache.Write(jsonBytes)
-	processModuleConfigCache.Close()
-	return err
+
+	if err := processModuleConfigCache.Close(); err != nil {
+		log.Info("Error closing file after writing processModuleConfigCache")
+		provisioner.removeProcessModuleConfigCache(tenantUUID)
+		return err
+	}
+	return nil
+}
+
+func (provisioner *OneAgentProvisioner) removeProcessModuleConfigCache(tenantUUID string) {
+	err := provisioner.fs.Remove(provisioner.path.AgentRuxitProcResponseCache(tenantUUID))
+	if os.IsNotExist(err) {
+		log.Info("processModuleConfigCache can't be removed, does not exist")
```

Comment:
If it does not exist it has the same effect as if it was removed, therefore I think this log message is not needed